common/test/kotlin/com/couchbase/lite/DeprecatedConfigFactoryTest.kt

```diff
@@ -0,0 +1,130 @@
+        ReplicatorConfigurationFactory.create(testDatabase, testEndpoint)
+    }
+
+    // Create with db and endpoint should succeed
+    @Test
+    fun testReplicatorConfigWithGoodArgs() {
+        val config = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint)
+        assertEquals(testDatabase, config.database)
+        assertEquals(testEndpoint, config.target)
+    }
+
+    // Create should copy source
+    @Test
+    fun testReplicatorConfigCopy() {
+        val config1 = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint, type = ReplicatorType.PULL)
+        val config2 = config1.create()
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(config1.target, config2.target)
+        assertEquals(config1.type, config2.type)
+    }
+
+    // Create should replace source
+    @Test
+    fun testReplicatorConfigReplace() {
+        val config1 = ReplicatorConfigurationFactory.create(testDatabase, testEndpoint, type = ReplicatorType.PULL)
+        val config2 = config1.create(type = ReplicatorType.PUSH)
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(config1.target, config2.target)
+        assertEquals(ReplicatorType.PUSH, config2.type)
+    }
+
+    // Create from a source explicitly specifying a default collection
+    @Test
+    fun testReplicatorConfigFromCollectionWithDefault() {
+        val config1 = ReplicatorConfigurationFactory
+            .newConfig(testEndpoint, mapOf(testDatabase.defaultCollection!! to CollectionConfiguration()))
+        val config2 = config1.create()
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        assertEquals(setOf(testCollection.database.defaultCollection), config2.collections)
+    }
+
+    // Create from a source with default collection, explicitly specifying a non-default collection
+    @Test
+    fun testReplicatorConfigFromCollectionWithDefaultAndOther() {
+        val config1 = ReplicatorConfigurationFactory
+            .newConfig(testEndpoint, mapOf(testCollection to CollectionConfiguration()))
+        val config2 = config1.create()
+        assertNotSame(config1, config2)
+        assertEquals(config1.database, config2.database)
+        // Information got lost here: should be a log message
+        assertEquals(setOf(testCollection.database.defaultCollection), config2.collections)
+    }
+
+    // Create from a source without default collection,
+    // explicitly specifying a non-default collection should fail
+    @Test(expected = IllegalArgumentException::class)
+    fun testReplicatorConfigFromCollectionWithoutDefault() {
+        testCollection.database.defaultCollection!!.delete()
+        val config1 = ReplicatorConfigurationFactory
+            .newConfig(testEndpoint, mapOf(testCollection to CollectionConfiguration()))
+        config1.create()
```

Comment:
I think I'm confused here. Why an exception is thrown here when there is no default collection?

Reply:
In this case, the database in config1 does not have a default collection.  It is not possible to configure, e.g., a document ID filter using ReplicatorConfiguration.create, because that attempts to configure the document ID filter on the default collection.  Unlike the situation above, where the default collection could be configured, here it cannot be configured.  It must be an error.